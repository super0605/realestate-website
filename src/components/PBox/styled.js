import styled from "styled-components";

export const PBoxContainer = styled.div`
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.06), 0 1px 0 0 rgba(0, 0, 0, 0.07),
    0 2px 1px -1px rgba(0, 0, 0, 0.06);
  background-color: ${({ theme }) => theme.colors.white};
  padding: ${({ padding }) => padding};
  margin-top: ${({ mTop }) => mTop};
  margin-bottom: ${({ mBottom }) => mBottom};
  margin-left: ${({ mLeft }) => mLeft};
  margin-right: ${({ mRight }) => mRight};
  padding-top: ${({ pTop }) => pTop};
  padding-bottom: ${({ pBottom }) => pBottom};
  padding-left: ${({ pLeft }) => pLeft};
  padding-right: ${({ pRight }) => pRight};
`;
