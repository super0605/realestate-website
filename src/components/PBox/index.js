import React from "react";
import PropTypes from "prop-types";
import { PBoxContainer } from "./styled";

const PBox = ({
  padding,
  mLeft,
  mRight,
  mTop,
  mBottom,
  pLeft,
  pRight,
  pTop,
  pBottom,
  children,
  ...props
}) => {
  return (
    <PBoxContainer
      padding={padding}
      mLeft={mLeft}
      mRight={mRight}
      mTop={mTop}
      mBottom={mBottom}
      pLeft={pLeft}
      pRight={pRight}
      pTop={pTop}
      pBottom={pBottom}
      {...props}
    >
      {children}
    </PBoxContainer>
  );
};

PBox.propTypes = {
  padding: PropTypes.string,
  children: PropTypes.node,
  mLeft: PropTypes.string,
  mRight: PropTypes.string,
  mTop: PropTypes.string,
  mBottom: PropTypes.string,
  pLeft: PropTypes.string,
  pRight: PropTypes.string,
  pTop: PropTypes.string,
  pBottom: PropTypes.string,
};

export default PBox;
