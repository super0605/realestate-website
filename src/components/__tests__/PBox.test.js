import * as React from "react";
import { mount, shallow } from "enzyme";
import PBox from "../PBox";

describe("PBox Component", () => {
  let props;
  let mountedPBox;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedPBox) {
        mountedPBox = bShallow
        ? shallow(<PBox {...props} />)
        : mount(<PBox {...props} />);
    }
    return mountedPBox;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
