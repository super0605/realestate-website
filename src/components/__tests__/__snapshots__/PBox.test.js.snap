// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PBox Component should render without problems 1`] = `<styled.div />`;
